staging/src/k8s.io/client-go/tools/cache/controller.go

```diff
@@ -336,6 +336,16 @@ func DeletionHandlingMetaNamespaceKeyFunc(obj interface{}) (string, error) {
 	return MetaNamespaceKeyFunc(obj)
 }
 
+// DeletionHandlingObjectToName checks for
+// DeletedFinalStateUnknown objects before calling
+// ObjectToName.
+func DeletionHandlingObjectToName(obj interface{}) (ObjectName, error) {
+	if d, ok := obj.(DeletedFinalStateUnknown); ok {
+		return ParseObjectName(d.Key)
```

Comment:
I could have gone the other way here, `return ObjectToName(d.Obj)`, but chose to parse `d.Key` instead because I think that is the well-trodden path, and I was not sure whether `d.Obj` is non-nil in all cases.